compiler/rustc_passes/src/check_attr.rs

```diff
@@ -451,17 +451,32 @@ impl CheckAttrVisitor<'tcx> {
         item: Option<&Item<'_>>,
         hir_id: HirId,
     ) {
+        let attrs: Vec<_> =
+            attrs.iter().filter(|attr| self.tcx.sess.check_name(attr, sym::repr)).collect();
+
+        if matches!(target, Target::Closure | Target::Expression | Target::Statement) {
```

Comment:
If feels somewhat weird to me to use a block list here, I think explicitly allowing `struct`s, `enum`s and `unions` would make more sense here.

Reply:
Fixed!